sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -17,135 +17,130 @@
 
 package org.apache.spark.sql
 
-import org.apache.spark.{SparkConf, SparkIllegalArgumentException}
-import org.apache.spark.sql.catalyst.expressions.{ExpressionEvalHelper, Literal, StringTrim, StringTrimLeft, StringTrimRight}
-import org.apache.spark.sql.catalyst.util.CollationFactory
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.test.SharedSparkSession
-import org.apache.spark.sql.types.{ArrayType, BinaryType, BooleanType, DataType, IntegerType, StringType}
+import org.apache.spark.sql.types._
 
 // scalastyle:off nonascii
 class CollationStringExpressionsSuite
   extends QueryTest
   with SharedSparkSession
   with ExpressionEvalHelper {
 
-  test("Support ConcatWs string expression with collation") {
-    // Supported collations
-    case class ConcatWsTestCase[R](s: String, a: Array[String], c: String, result: R)
+  test("Support `ConcatWs` string expression with collation") {
+    case class ConcatWsTestCase[R](
+        sep: String,
+        arrayStr: Array[String],
+        collation: String,
+        result: R)
     val testCases = Seq(
       ConcatWsTestCase(" ", Array("Spark", "SQL"), "UTF8_BINARY", "Spark SQL"),
       ConcatWsTestCase(" ", Array("Spark", "SQL"), "UTF8_LCASE", "Spark SQL"),
       ConcatWsTestCase(" ", Array("Spark", "SQL"), "UNICODE", "Spark SQL"),
       ConcatWsTestCase(" ", Array("Spark", "SQL"), "UNICODE_CI", "Spark SQL")
     )
     testCases.foreach(t => {
-      val arrCollated = t.a.map(s => s"collate('$s', '${t.c}')").mkString(", ")
-      var query = s"SELECT concat_ws(collate('${t.s}', '${t.c}'), $arrCollated)"
-      // Result & data type
-      checkAnswer(sql(query), Row(t.result))
-      assert(sql(query).schema.fields.head.dataType.sameType(StringType(t.c)))
-      // Implicit casting
```

Comment:
Did you remove the implicit casting tests? If so, please, explain why.

Reply:
As I mentioned in the comment above, these tests will be moved to the `collations.sql` golden file, as part of a separate PR